pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/kubernetes/KubernetesRuntime.java

```diff
@@ -973,8 +982,19 @@ private V1PodSpec getPodSpec(List<String> instanceCommand, Function.Resources re
     V1Container getFunctionContainer(List<String> instanceCommand, Function.Resources resource) {
         final V1Container container = new V1Container().name(PULSARFUNCTIONS_CONTAINER_NAME);
 
-        // set up the container images
-        container.setImage(pulsarDockerImageName);
+        // By default, if we do not use the function image of a specific language, then we will support all languages.
+        if (javaFunctionDockerImageName == null && pythonFunctionDockerImageName == null && goFunctionDockerImageName == null) {
```

Comment:
I think the logic here is completely wrong. You should set the correct docker image name based on the language runtime, not based on whether this configuration is set or not.

Reply:
> I think the logic here is completely wrong.

In here, my processing idea uses the original `pulsarDockerImageName` as the default configuration, that is, when we do not set the corresponding image for the function of the specified language, we use the default image. 

> You should set the correct docker image name based on the language runtime, not based on whether this configuration is set or not.

Do you mean that we check the image that the user needs to use based on the parameters passed by the user `--jar`,` --python` or `--go`?

Reply:
You need to check the runtime of the function and then the image based on that.
`instanceConfig.getFunctionDetails().getRuntime()`